code/modules/reagents/reagent_containers/medipack.dm

```diff
@@ -0,0 +1,146 @@
+	name = "medium medipack"
+	desc = "This medium-sized pack of containers seems more effective at patching decently-sized wounds. A concotion of highly specialized 'miracle drugs', impossible to reproduce without specialized equipment. Knits flesh together in an instant."
+	icon_state = "medium"
+	w_class = WEIGHT_CLASS_NORMAL
+
+	blood_loss = 50
+	organ_loss = 0.5 // 2.5 health per organ.
+	heal_modifier = 0.2 // heals 20% of damage taken.
+	jelly_amount = 1 // used for a burst of healing after being healed.
+	heal_delay = 12.5 // 1.25 seconds.
+	flat_heal = 9
+
+/obj/item/medipack/large
+	name = "large medipack"
+	desc = "This large-sized pack of containers seems more effective at patching most wounds. A concotion of highly specialized 'miracle drugs', impossible to reproduce without specialized equipment. Knits flesh together in an instant."
+	icon_state = "large"
+	w_class = WEIGHT_CLASS_BULKY // requires a specialized container to store it.
+
+	blood_loss = 150
+	organ_loss = 2 // 10 health per organ.
+	heal_modifier = 0.45 // heals 45% of damage taken.
+	jelly_amount = 7.5 // used for a burst of healing after being healed.
+	heal_delay = 20 // 2 seconds.
+	self_heal_delay_modifier = 10 //Takes far longer to use on yourself, stick together.
+	flat_heal = 5
+
+/obj/item/medipack/large/proc/in_box_sprite()
+	return "[icon_state]_inbox"
+
+/obj/item/medipack/adrenaline
+	name = "adrenaline medipack"
+	desc = "This needle like container can revive the dead in a pinch. It takes time to apply, as incorrect use could kill the patient. A concotion of highly specialized 'miracle drugs', impossible to reproduce without specialized equipment. Knits flesh together in an instant."
+	icon_state = "needle"
+
+	blood_loss = 200
+	jelly_amount = 5 // used for a burst of healing after being healed.
+	flat_heal = 5
+	life_loss = TRUE // revives the dead
+
+/obj/item/storage/fancy/medipack
+	name = "medipack holder"
+	desc = "This specialized container can hold large medipacks and still fit in backpacks."
+	icon = 'necromorphs/icons/obj/medipack.dmi'
+	icon_state = "holder"
+	base_icon_state = "holder"
+	spawn_type = /obj/item/medipack/large
+	is_open = TRUE
+	appearance_flags = KEEP_TOGETHER|LONG_GLIDE
+	contents_tag = "large"
+
+	fold_result = /obj/item/stack/sheet/plastic
+
+/obj/item/storage/fancy/medipack/ComponentInitialize()
+	. = ..()
+	var/datum/component/storage/STR = GetComponent(/datum/component/storage)
+	STR.max_items = 1
+	STR.set_holdable(list(/obj/item/medipack/large))
+	STR.max_w_class = WEIGHT_CLASS_BULKY
+
+/obj/item/storage/fancy/medipack/PopulateContents()
+	. = ..()
+	update_appearance()
+
+/obj/item/storage/fancy/medipack/update_icon_state()
+	. = ..()
+	icon_state = "[base_icon_state]"
+
+/obj/item/storage/fancy/medipack/update_overlays()
+	. = ..()
+	if(!is_open)
+		return
+
+	var/medipacks = 0
+	for(var/obj/item/medipack/large/medipack in contents)
+
+		. += image(icon = initial(icon), icon_state = medipack.in_box_sprite())
```

Comment:
Use `mutable_appearance()` instead

Reply:
`in_box_sprite()` is only used here. Just inline it.